### PR TITLE
nimbus-verified-proxy integration

### PR DESCRIPTION
**Pull request by vitvly**
Tracking issue: https://github.com/status-im/status-go/issues/3282

status-go code depends on an external repo - `lc-proxy-wrapper`, that provides functions wrapping actual nimbus-eth1 code. Relevant functionality is toggled by a build tag.

### CODE REVIEW

File: rpc/verif_proxy.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+			LogLevel:   "INFO",
+			//Eth2Network:      "prater",
+			//TrustedBlockRoot: "0x017e4563ebf7fed67cff819c63d8da397b4ed0452a3bbd7cae13476abc5020e4",
+		}
+		proxy.StartLightClient(ctx, &testConfig)
+		verifProxy := NewVerifProxy(&testConfig)
+		c.RegisterHandler(
+			params.BalanceMethodName,
+			func(ctx context.Context, v uint64, params ...interface{}) (interface{}, error) {
+				fmt.Println("### inside registered handler")
```
Comment (cammellos) [inline rpc/verif_proxy.go]: println

File: rpc/verif_proxy.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+				addr := params[0].(common.Address)
+				return verifProxy.GetBalance(ctx, addr)
+			},
+		)
+	}
+}
+
+func NewVerifProxy(cfg *proxy.Config) *VerifProxy {
+	endpoint := "http://" + cfg.RpcAddress + ":" + fmt.Sprint(cfg.RpcPort)
+	fmt.Println("### NewVerifProxy: ", endpoint)
```
Comment (cammellos) [inline rpc/verif_proxy.go]: println

File: rpc/verif_proxy.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	proxy.client = client
+	return proxy
+}
+
+func (p *VerifProxy) GetBalance(ctx context.Context, address common.Address) (interface{}, error) {
+	fmt.Println("### IN GetBalance, address ", address)
+	var result hexutil.Big
+	err := p.client.CallContext(ctx, &result, "eth_getBalance", address, "latest")
+	if err != nil {
+		fmt.Println("### IN GetBalance, error ", err)
```
Comment (cammellos) [inline rpc/verif_proxy.go]: use logger

**Review by cammellos — APPROVED** [review]

**Comment by vitvly** [comment]
Thanks @cammellos 
  - comments removed
  - used `Client`'s logger
  - removed superfluous `Makefile` lines

File: rpc/verif_proxy.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
+	gethrpc "github.com/ethereum/go-ethereum/rpc"
+	proxy "github.com/siphiuel/lc-proxy-wrapper"
```
Comment (Samyoul) [inline rpc/verif_proxy.go:16]: There is a chance that this line will give a lint error as your own repos would be considered 3rd party and should be grouped with other 3rd party packages.

File: rpc/verif_proxy_test.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"testing"
+	"time"
+
+	"net/http"
+	"net/http/httptest"
```
Comment (Samyoul) [inline rpc/verif_proxy_test.go:16]: These are all std lib packages so can be grouped together

**Review by Samyoul — APPROVED** [review]
This looks good to me, nice work.

Two minor points:
- don't forget to bump VERSION
- do you need to run `make vendor`? Because I see dependency changes but no change to the vendor directory

**Comment by vitvly** [comment]
Thank you @Samyoul , done.

**Pull request merged**